### PR TITLE
Add Github action to build Docker images

### PR DESCRIPTION
**Pull request by alex9smith**
So we don't merge anything that breaks the image. 

This just tests that the images can be built successfully, but I think that should be sufficient as a pre-merge check. It's pretty tough to run the base images on their own without an app to check that eg. nginx starts, the image serves some html etc, and I'm not sure it's worth the effort.

https://trello.com/c/iFGhnPkx/2277-frontend-docker-base-image-isnt-building-with-node-14173

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        uses: actions/checkout@v2
+
+      - name: Set common environment variables
+        run: |
+          echo "build_date=$(date -u '+%Y%m%dT%H%M%SZ')" >> $GITHUB_ENV
+          echo "build_version=$(cat VERSION)" >> $GITHUB_ENV
+
+      - name: Build Docker base image
+        run: |
+          docker pull digitalmarketplace/base
```
Comment (bjgill) [inline .github/workflows/test.yml:18]: Why do we need to explicitly pull this image?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+          docker build --pull --cache-from digitalmarketplace/base --build-arg BUILD_DATE=${{ env.build_date }} --build-arg BUILD_VERSION=${{ env.build_version }} -t digitalmarketplace/base -f base.docker .
+
+      - name: Build API image
+        run: docker build -f api.docker .
```
Comment (bjgill) [inline .github/workflows/test.yml:22]: Nit: would make it easier to review/understand with the long version of options (`--file`)

**Review by bjgill — APPROVED** [review]
Good idea - should make it harder for me to break the build next time

Just two minor questions - feel free to merge when ready.

File: .github/workflows/test.yml (hunk 1 above)
Comment (alex9smith) [reply .github/workflows/test.yml:18]: It's so in the next command we can build using `--cache-from` and potentially save some time. The Docker docs aren't very clear on this point , but Gitlab has it in [theirs](https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#how-docker-caching-works) for how to speed up Docker builds on CI

File: .github/workflows/test.yml (hunk 2 above)
Comment (alex9smith) [reply .github/workflows/test.yml]: Good point, added in bd13de8

**Pull request merged**